common/data/src/main/java/org/thingsboard/server/common/data/EdgeUtils.java

```diff
@@ -25,60 +25,35 @@
 import org.thingsboard.server.common.data.id.EntityId;
 import org.thingsboard.server.common.data.id.TenantId;
 
+import java.util.EnumMap;
 import java.util.concurrent.ThreadLocalRandom;
 
 @Slf4j
 public final class EdgeUtils {
 
-    private static final int STACK_TRACE_LIMIT = 10;
-
-    private EdgeUtils() {
-    }
+    private static final EnumMap<EntityType, EdgeEventType> entityTypeEdgeEventTypeEnumMap;
 
-    public static EdgeEventType getEdgeEventTypeByEntityType(EntityType entityType) {
-        switch (entityType) {
-            case EDGE:
-                return EdgeEventType.EDGE;
-            case DEVICE:
-                return EdgeEventType.DEVICE;
-            case DEVICE_PROFILE:
-                return EdgeEventType.DEVICE_PROFILE;
-            case ASSET:
-                return EdgeEventType.ASSET;
-            case ASSET_PROFILE:
-                return EdgeEventType.ASSET_PROFILE;
-            case ENTITY_VIEW:
-                return EdgeEventType.ENTITY_VIEW;
-            case DASHBOARD:
-                return EdgeEventType.DASHBOARD;
-            case USER:
-                return EdgeEventType.USER;
-            case RULE_CHAIN:
-                return EdgeEventType.RULE_CHAIN;
-            case ALARM:
-                return EdgeEventType.ALARM;
-            case TENANT:
-                return EdgeEventType.TENANT;
-            case CUSTOMER:
-                return EdgeEventType.CUSTOMER;
-            case WIDGETS_BUNDLE:
-                return EdgeEventType.WIDGETS_BUNDLE;
-            case WIDGET_TYPE:
-                return EdgeEventType.WIDGET_TYPE;
-            case OTA_PACKAGE:
-                return EdgeEventType.OTA_PACKAGE;
-            case QUEUE:
-                return EdgeEventType.QUEUE;
-            default:
-                log.warn("Unsupported entity type [{}]", entityType);
-                return null;
+    static {
+        entityTypeEdgeEventTypeEnumMap = new EnumMap<>(EntityType.class);
+        for (EdgeEventType edgeEventType : EdgeEventType.values()) {
+            if (edgeEventType.getEntityType() != null) {
+                entityTypeEdgeEventTypeEnumMap.put(edgeEventType.getEntityType(), edgeEventType);
+            }
         }
     }
 
+    private static final int STACK_TRACE_LIMIT = 10;
+
+    private EdgeUtils() {}
+
     public static int nextPositiveInt() {
         return ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
     }
 
+    public static EdgeEventType getEdgeEventTypeByEntityType(EntityType entityType) {
+        return entityTypeEdgeEventTypeEnumMap.get(entityType);
```

Comment:
You have missed the warning after refactoring. If map does not have the value, you should print the same warnings.
See:

```
log.warn("Unsupported entity type [{}]", entityType);
```

Reply:
Thanks for noticing, but I removed this warning on purpose from being duplicated. At the method's invocation points, there were (and still are) null checkers, which log if the type is not supported. If this warning needs to be returned as well, I will provide it. 

For example, call points
![image](https://github.com/thingsboard/thingsboard/assets/50847617/b21f95bb-7e4d-4c46-8c4e-126ee9365822)
![image](https://github.com/thingsboard/thingsboard/assets/50847617/a78ed735-3b19-4ce1-bf60-a9537b56d9a4)